decimal.go

```diff
@@ -1224,14 +1224,32 @@ func (d Decimal) Ln(precision int32) (Decimal, error) {
 }
 
 // NumDigits returns the number of digits of the decimal coefficient (d.Value)
-// Note: Current implementation is extremely slow for large decimals and/or decimals with large fractional part
 func (d Decimal) NumDigits() int {
-	d.ensureInitialized()
-	// Note(mwoss): It can be optimized, unnecessary cast of big.Int to string
-	if d.IsNegative() {
-		return len(d.value.String()) - 1
+	if d.value == nil {
+		return 1
+	}
+
+	if d.value.IsInt64() {
+		i64 := d.value.Int64()
+		if i64 <= (1<<53) && i64 >= -(1<<53) {
```

Comment:
this range could be extended looking for the first int64 whose inexact float64 doesn't have same number of digits, but decided to keep it simple

Reply:
Agree, let's keep it simple. New implementation is already a lot faster than the old one and I'm happy with it :D 
I would only leave a comment for future contributors that 1 << 53 represents the significand precision of floating-point numbers.

Reply:
Added comment